app/javascript/packs/components/Punches.js

```diff
@@ -1,13 +1,28 @@
-import React from 'react';
-import Punch from './Punch';
+import React from "react";
+import Punch from "./Punch";
+import { isEmpty } from "lodash";
 
 const Punches = ({ sheet }) => {
+  const workedHours = () => {
+    const total_delta = sheet.reduce(
+      (acc, current) => (acc += current.delta),
+      0
+    );
+
+    return `Total: ${total_delta}h`;
+  };
+
   return (
-    <ul className="punches">
-      { sheet.map((punch, i) =>
-       <Punch key={i} punch={punch} />
+    <>
+      {!isEmpty(sheet) && (
+        <ul className="punches">
+          {sheet.map((punch, i) => (
+            <Punch key={i} punch={punch} />
+          ))}
+          <li>{workedHours(sheet)}</li>
```

Comment:
Here you're passing an argument to the function but the function doesn't use this argument. You can either use the function without passing a variable, defining the value as a derived state, or moving the function outside of the component. I personally prefer the latter.
```js
const workedHours = (sheet) => {
  // ...
}

const Punches = ({ sheet }) => {
  return <>
      ...
      <li>{workedHours(sheet)}</li>
      ...
  </>
}
```